File: CS/ViewModel/AutoHideViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Docking;

namespace WpfApplication {
    public class AutoHideViewModel : IMVVMDockingProperties {
        public AutoHideViewModel(string caption) {
            Caption = caption;
            MyTargetName = "auotHideGroup";
        }
        public string Caption { get; set; }
        public string MyTargetName { get; set; }
        string IMVVMDockingProperties.TargetName {
            get {
                return MyTargetName;
            }
            set {
                MyTargetName = value;
            }
        }
    }
}

File: VB/ViewModel/AutoHideViewModel.vb
Imports DevExpress.Xpf.Docking

Namespace WpfApplication

    Public Class AutoHideViewModel
        Implements IMVVMDockingProperties

        Public Sub New(ByVal caption As String)
            Me.Caption = caption
            MyTargetName = "auotHideGroup"
        End Sub

        Public Property Caption As String

        Public Property MyTargetName As String

        Private Property TargetName As String Implements IMVVMDockingProperties.TargetName
            Get
                Return MyTargetName
            End Get

            Set(ByVal value As String)
                MyTargetName = value
            End Set
        End Property
    End Class
End Namespace
